### PR TITLE
Closes #208 208_exprs@devel: Change `vars()` to `exprs()`

### LINKED ISSUE #208: Replace `vars()` with `exprs()`
### Background Information

`vars()` should be replaced by `exprs()` in all admiral packages (see https://github.com/pharmaverse/admiral/issues/1627).

- assertions need to be updated
- any other functions expecting a list of variables need to be updated
- programming strategy needs to be updated

### Definition of Done

_No response_

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
Do you think we need deprecations warnings for our `admiraldev` package - since it is really only for us?  Just one less thing for us to maintain.

I'm also uncertain if `admiraldev 0.4` will be backwards compatible with `admiral 0.9.0`??

**Comment by bundfussr** [comment]
@bms63 , I assumed we would use the same deprecation strategy as for admiral core.

Most likely we are the only ones who are using admiraldev at the moment. So maybe we do not need the deprecation for this release.

In future I would expect that developers of TA-, company-, or project-specific admiral packages are using admiraldev as well. These may not be in the admiral dev team. Thus using our deprecation strategy for admiraldev could be helpful for them.

@thomas-neitmann , what do you think?

**Comment by bundfussr** [comment]
admiraldev 0.3.0 will not be compatible with admiral 0.9.0.

**Comment by bms63** [comment]
Umm..then we should follow the strategy even for this release just so we get in the habit.  I think we could use a github action to deploy this as a new issue, i.e. following our deprecation strategy,  for when we do a release.

**Comment by bundfussr** [comment]
@bms63 , what do you mean by
> I think we could use a github action to deploy this as a new issue, i.e. following our deprecation strategy, for when we do a release.

???

**Comment by bms63** [comment]
> @bms63 , what do you mean by
> 
> > I think we could use a github action to deploy this as a new issue, i.e. following our deprecation strategy, for when we do a release.
> 
> ???

Like we do for the CRAN Status Monitor - the action makes a new issue automatically if an error is detected on CRAN.  For this case, when we do a release an auto-generated issue is created reminding us to deprecate.  Be really cool if it could scrape our code for deprecated functions and list them out.

**Comment by bundfussr** [comment]
@bms63 , I see. This seems to be similar to https://github.com/pharmaverse/admiraldev/issues/36.

File: tests/testthat/test-quo.R [hunk 1]
```diff
@@ -23,18 +23,44 @@ test_that("quo_c Test 2: `quo_c` returns error if non-quosures are input", {
   )
```
Comment (zdz2101) [inline tests/testthat/test-quo.R:23]: Similar to the aforementioned commentary about the deprecation process. 

For Test 1 and Test 2 above, do we need include the usual `class = "lifecycle_warning_deprecated"` part?

Comment (bundfussr) [reply tests/testthat/test-quo.R:31]: I have added a test for the deprecation warning.

**Review by zdz2101 — APPROVED** [review]
